internal/api/cluster_installation_test.go

```diff
@@ -743,6 +743,170 @@ func TestMigrateDNS(t *testing.T) {
+		DNS:       "dns3.example.com",
+		Database:  model.InstallationDatabaseMysqlOperator,
+		Filestore: model.InstallationFilestoreMinioOperator,
+		Affinity:  model.InstallationAffinityIsolated,
+		State:     model.InstallationStateHibernating,
+	}
+
+	err = sqlStore.CreateInstallation(installation3, nil)
+	require.NoError(t, err)
+
+	time.Sleep(1 * time.Millisecond)
+	clusterInstallation1 := &model.ClusterInstallation{
+		ClusterID:      sourceCluster.ID,
+		InstallationID: installation1.ID,
+		Namespace:      "namespace_10",
+		State:          model.ClusterInstallationStateCreationRequested,
+	}
+	err = sqlStore.CreateClusterInstallation(clusterInstallation1)
+	require.NoError(t, err)
+
+	clusterInstallation2 := &model.ClusterInstallation{
+		ClusterID:      sourceCluster.ID,
+		InstallationID: installation2.ID,
+		Namespace:      "namespace_11",
+		State:          model.ClusterInstallationStateCreationRequested,
+	}
+	err = sqlStore.CreateClusterInstallation(clusterInstallation2)
+	require.NoError(t, err)
+
+	clusterInstallation3 := &model.ClusterInstallation{
+		ClusterID:      sourceCluster.ID,
+		InstallationID: installation3.ID,
+		Namespace:      "namespace_11",
+		State:          model.ClusterInstallationStateCreationRequested,
+	}
+	err = sqlStore.CreateClusterInstallation(clusterInstallation3)
+	require.NoError(t, err)
+
+	targetCluster, err := client.CreateCluster(&model.CreateClusterRequest{
+		Provider:    model.ProviderAWS,
+		Zones:       []string{"zone"},
+		Annotations: []string{"my-annotation"},
+	})
+	require.NoError(t, err)
+	require.NotNil(t, targetCluster)
+
+	t.Run("valid migration test", func(t *testing.T) {
+		err := client.MigrateClusterInstallation(&model.MigrateClusterInstallationRequest{InstallationID: "", SourceClusterID: sourceCluster.ID, TargetClusterID: targetCluster.ID, DNSSwitch: false, LockInstallation: false})
+		require.NoError(t, err)
+
+		err = client.MigrateDNS(&model.MigrateClusterInstallationRequest{InstallationID: "", SourceClusterID: sourceCluster.ID, TargetClusterID: targetCluster.ID, DNSSwitch: true, LockInstallation: true})
+		require.NoError(t, err)
+
+		// varifying the outcomes
+		var isActiveClusterInstallations = false
+		filter := &model.ClusterInstallationFilter{
+			ClusterID:      sourceCluster.ID,
+			InstallationID: "",
+			Paging:         model.AllPagesNotDeleted(),
+			IsActive:       &isActiveClusterInstallations,
+		}
+		cis, err := sqlStore.GetClusterInstallations(filter)
+		require.NoError(t, err)
+		require.NotEmpty(t, cis)
+
+		isActiveClusterInstallations = true
+		filter = &model.ClusterInstallationFilter{
+			ClusterID:      targetCluster.ID,
+			InstallationID: "",
+			Paging:         model.AllPagesNotDeleted(),
+			IsActive:       &isActiveClusterInstallations,
+		}
+		cis, err = sqlStore.GetClusterInstallations(filter)
+		require.NoError(t, err)
+		require.NotEmpty(t, cis)
+		assert.Len(t, cis, 3)
+	})
+
+}
+func TestMigrateDNSForHibernatingInstallation(t *testing.T) {
```

Comment:
Are those tests `TestMigrateDNSForHibernatingInstallation` and `TestMigrateDNSForNonHibernatingInstallation` covering anything more than `TestMigrateDNS`? It seems that in `TestMigrateDNS` we already create both non-hibernating and hibernating installations. If so I would opt for deleting them to reduce the amount of maintenance required.

Reply:
During migration i came across 3 possible use cases.

1. - Migrating everything
2. - Migrating only hibernating installations 
3. - Migrating only non hibernating installations

e.g the locking fix in this PR can easily be identified earlier if we already had test coverage for above point 2 & 3. Although we r good for now but i think keeping them for future coverage would be good.